src/Tests/MPQEditor/MPQEditor.test.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+
+import { assert } from "chai";
+import { MPQEditorProvider } from "../../MPQEditor/MPQEditor";
+import { TestBuilder } from "../TestBuilder";
+
+import { ExtensionContext, extensions } from "vscode";
+
+suite("MPQEditorProvider", function () {
+  let extensionContext: ExtensionContext;
+  let testBuilder: TestBuilder;
+
+  suiteSetup(async () => {
+    // Trigger extension activation and grab the context as some tests depend on it
+    const extensionId = "Samsung.one-vscode";
+    await extensions.getExtension(extensionId)?.activate();
+    extensionContext = (global as any).extensionContext;
+  });
+
+  setup(() => {
+    testBuilder = new TestBuilder(this);
+    testBuilder.setUp();
+  });
+
+  teardown(() => {
+    testBuilder.tearDown();
+  });
+  suite("#constructor()", function () {
+    test("is constructed", function () {
+      const provider = new MPQEditorProvider(extensionContext);
+      assert.instanceOf(provider, MPQEditorProvider);
+    });
+  });
+
+  suite("#findMPQName", function () {
+    test("test findMPQName", function () {
+      const baseMPQName: string = "model";
+      const dirPath: string = testBuilder.dirInTemp;
+
+      const content = `
+            empty content
+            `;
+
+      let targetNames: string[] = [
+        "model.mpq.json",
+        "model(1).mpq.json",
+        "model(1)(1).mpq.json",
```

Comment:
I wonder why you have decided to generate model name as such?
I think it's rather natural that when model.mpq.json and model(1).mpq.json exists, model(2).mpq.json be the next..

Reply:
@dayo09 
Thank you.
Please see https://github.com/Samsung/ONE-vscode/pull/1505#issuecomment-1437715625.
Just adding `(1)` is easy to implement, but i can change it to another scheme if needed.